content-services/content-sources-backend
# Fixes 2965: add create/fetch/list for templates

## Summary

- Adds create, list, and fetch endpoints for content templates
- Adds filters and sorting for name, version, and arch
- Adds search by name

## Testing steps
1. Create a repository
2. Create templates

```
POST http://localhost:8000/api/content-sources/v1.0/templates/
Content-Type: application/json
x-Rh-Identity: eyJpZGVudGl0eSI6eyJ0eXBlIjoiVXNlciIsInVzZXIiOnsidXNlcm5hbWUiOiJzbmFwVXNlciJ9LCJhY2NvdW50X251bWJlciI6IjE2Nzc5ODUyIiwiaW50ZXJuYWwiOnsib3JnX2lkIjoiMTY3Nzk4NTIifX19Cg==

{
  "name": "my template 5",
  "repository_uuids": ["8e94a78e-915e-4eee-8781-7592b5e3f6f4"],
  "description": "my new template",
  "arch": "x86_64",
  "version": "7"

}
```
3. List Templates with filtering, sorting, or searching
```
GET http://localhost:8000/api/content-sources/v1.0/templates/?version=7&sort_by=name
Content-Type: application/json
x-Rh-Identity: eyJpZGVudGl0eSI6eyJ0eXBlIjoiVXNlciIsInVzZXIiOnsidXNlcm5hbWUiOiJzbmFwVXNlciJ9LCJhY2NvdW50X251bWJlciI6IjE2Nzc5ODUyIiwiaW50ZXJuYWwiOnsib3JnX2lkIjoiMTY3Nzk4NTIifX19Cg==
```
4. Fetch Templates
```
GET http://localhost:8000/api/content-sources/v1.0/templates/7fc1ee4a-fad2-4e4c-9979-953136c6eb09
Content-Type: application/json
x-Rh-Identity: eyJpZGVudGl0eSI6eyJ0eXBlIjoiVXNlciIsInVzZXIiOnsidXNlcm5hbWUiOiJzbmFwVXNlciJ9LCJhY2NvdW50X251bWJlciI6IjE2Nzc5ODUyIiwiaW50ZXJuYWwiOnsib3JnX2lkIjoiMTY3Nzk4NTIifX19Cg==
```
5. Make sure to look at the JIRA card and check against the requirements there in case I missed something.


## Checklist
